app/models/occupation_standard.rb

```diff
@@ -377,6 +377,22 @@ def state_id
     state&.id
   end
 
+  def clean_onet_code
+    # correct format 12-3456.07
+    base_onet = onet_code
+
+    if onet_code.present?
+      return nil if onet_code == "onet"
+
+      base_onet = base_onet.delete("^0-9")
+      return onet_code if base_onet.length < 8
```

Comment:
```suggestion
      return onet_code if base_onet.length < 7
```

You're trying to skip the ones like "12-3456", so wouldn't it be less than 7, since at this line, base_onet will look like: "123456" ?

Reply:
The expected length is 8. I don't know if the ones that are 7 need extra care too. 12-3456 is 2 digits shorter than the other onets.